source/client/stream_decoder.cc

```diff
@@ -125,23 +125,29 @@ void StreamDecoder::onPoolReady(Envoy::Http::RequestEncoder& encoder,
                           "HTTP headers in {}.",
                           *request_headers_);
   }
-  if (request_body_size_ > 0) {
+  if (request_body_size_ > 0 || !request_body_.empty()) {
```

Comment:
It looks like the `addBytesReceived` call below should do something different in the two cases.